ref/go/src/bech32/bech32_test.go

```diff
@@ -0,0 +1,250 @@
+// Copyright (c) 2017 Takatoshi Nakagawa
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package bech32
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+	"testing"
+)
+
+func segwit_scriptpubkey(version int, program []int) []int {
+	if version != 0 {
+		return append(append([]int{version + 0x80}, len(program)), program...)
+	} else {
+		return append(append([]int{version}, len(program)), program...)
+
+	}
+}
+
+var VALID_CHECKSUM []string = []string{
+	"A12UEL5L",
+	"an83characterlonghumanreadablepartthatcontainsthenumber1andtheexcludedcharactersbio1tt5tgs",
+	"abcdef1qpzry9x8gf2tvdw0s3jn54khce6mua7lmqqqxw",
+	"11qqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqc8247j",
+	"split1checkupstagehandshakeupstreamerranterredcaperred2y9e3w",
+}
+
+type item struct {
+	address      string
+	scriptpubkey []int
+}
+
+var VALID_ADDRESS []item = []item{
+	item{"BC1QW508D6QEJXTDG4Y5R3ZARVARY0C5XW7KV8F3T4",
+		[]int{
+			0x00, 0x14, 0x75, 0x1e, 0x76, 0xe8, 0x19, 0x91, 0x96, 0xd4, 0x54,
```

Comment:
It would be better to keep those as the original hex values and decode the hex inside the test. If more test vectors are added, it would be easier to add them here.